keanu-project/src/main/java/io/improbable/keanu/vertices/VertexLabel.java

```diff
@@ -1,30 +1,58 @@
 package io.improbable.keanu.vertices;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import java.util.Objects;
 
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+
 public class VertexLabel {
-    private final String label;
+    private static final char NAMESPACE_SEPARATOR = '.';
+    private static final char NAME_PREPENDER = ':';
```

Comment:
Obviously not something we care about too much at the moment, but it's definitely the case that when we come to read this stuff in we'll want to avoid our separators being part of the actual identifiers in the namespace.  It might be worth checking for these things now to ensure that any users of this functionality don't hit issues down the line if they use an invalid character?

Reply:
these constants are only used in `toString`

Reply:
Yeah, that doesn't mean it couldn't lead to confusion though.

If, say, a Keanu user started using Vertex Labels (which they will if they use Graphception) then they could very well accidentally use a character we've decided to use as a namespace separator meaning both:
A) We could get confusing toString output (which could be important for both internal debug and potential visualisation/editor tooling).
B) If we want to save & load Vertices to disk then  either we make their existing Vertex Labels invalid, or we need to use a more complex encoding scheme for names when storing them to disk.

So, not stopping them using the separator characters now could well cause them headaches/confusion in future if we add extra functionality.  Seems worth the small amount of extra code to me for now.

Thinking more deeply on this, there are other potential characters that would cause issues for case B - (eg special characters for JSON) - so longer term we will probably want to vet any string they pass, but perhaps we just need to raise a new ticket for this.

Reply:
ok, so in conclusion shall we go with:
* remove `NAME_PREPENDER` and use `NAMESPACE_SEPARATOR` in its place
* set `NAMESPACE_SEPARATOR` to be `/` instead of `.`
* throw if the user passes in a string containing `NAMESPACE_SEPARATOR `

Reply:
That sounds perfect to me - thanks for thinking about this!